model/model.go

```diff
@@ -94,6 +95,28 @@ func New(s string) (Model, error) {
 	return m, nil
 }
 
+func NewTextProcessor(s string) (TextProcessor, error) {
+	r, err := os.Open(s)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+	meta, _, err := fs.Decode(r, -1)
+	if err != nil {
+		return nil, err
+	}
+	arch := meta.KV().Architecture()
+	f, ok := models[arch]
+	if !ok {
+		return nil, fmt.Errorf("unsupported model architecture %q", arch)
+	}
+	m, err := f(meta.KV())
+	if err != nil {
+		return nil, err
+	}
+	return m.(TextProcessor), nil
```

Comment:
Is there a clean way to avoid the casting here?

Reply:
I don't think so, but I'll improve the error handling for models that don't have text processors.